apps/pkey.c

```diff
@@ -277,8 +277,10 @@ int pkey_main(int argc, char **argv)
             BIO_printf(out, "Key is invalid\n");
 
             while ((err = ERR_peek_error()) != 0) {
-                BIO_printf(out, "Detailed error: %s\n",
-                           ERR_reason_error_string(err));
+                const char *reason = ERR_reason_error_string(err);
+
+                if (reason != NULL)
+                    BIO_printf(out, "Detailed error: %s\n", reason);
                 ERR_get_error(); /* remove err from error stack */
             }
```

Comment:
Why do we have this loop at all?  Just replace it with this (*in the interest of making our apps consistent!*):

``` C
            ret = 0;
            ERR_print_errors(bio_err);
```

While you're at it, please make sure that all error message go to `bio_err` instead of `out`.

Reply:
Ok, I'll do.
Yet are you sure about adding `ret = 0;` here? This means EXIT_SUCCESS.

Reply:
> Yet are you sure about adding `ret = 0;` here? This means EXIT_SUCCESS.

Ah, right, sorry, I confused myself.

Reply:
Done (except for the assignment `ret = 0`).